src/main/java/org/prebid/server/bidder/richaudience/RichaudienceBidder.java

```diff
@@ -0,0 +1,176 @@
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+import org.prebid.server.util.ObjectUtil;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class RichaudienceBidder implements Bidder<BidRequest> {
+
+    private static final String DEFAULT_CURRENCY = "USD";
+    private static final TypeReference<ExtPrebid<?, ExtImpRichaudience>> RICHAUDIENCE_EXT_TYPE_REFERENCE =
+            new TypeReference<>() {
+            };
+
+    private final String endpointUrl;
+    private final JacksonMapper mapper;
+
+    public RichaudienceBidder(String endpointUrl, JacksonMapper mapper) {
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+        this.mapper = Objects.requireNonNull(mapper);
+    }
+
+    @Override
+    public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest request) {
+        final URL url = extractUrl(request);
+        final boolean isSecure = "https".equals(ObjectUtil.getIfNotNull(url, URL::getProtocol));
+        final List<Imp> modifiedImps = new ArrayList<>();
+        boolean isTest = false;
+
+        try {
+            validateRequest(request);
+            for (Imp imp : request.getImp()) {
+                validateImp(imp);
+
+                final ExtImpRichaudience extImpRichaudience = parseImpExt(imp);
+                modifiedImps.add(modifyImp(imp, extImpRichaudience, isSecure));
+
+                isTest |= BooleanUtils.isTrue(extImpRichaudience.getTest());
+            }
+        } catch (PreBidException e) {
+            return Result.withError(BidderError.badInput(e.getMessage()));
+        }
+
+        final BidRequest modifiedRequest = modifyBidRequest(request, url, modifiedImps, isTest);
+        return Result.withValue(createHttpRequest(modifiedRequest));
+    }
+
+    private static void validateRequest(BidRequest bidRequest) throws PreBidException {
+        final Device device = bidRequest.getDevice();
+        final boolean isDeviceValid = device != null && !StringUtils.isAllBlank(device.getIp(), device.getIpv6());
+        if (!isDeviceValid) {
+            throw new PreBidException("Device IP is required.");
+        }
+    }
+
+    private static URL extractUrl(BidRequest bidRequest) {
+        try {
+            return new URL(ObjectUtil.getIfNotNull(bidRequest.getSite(), Site::getPage));
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    private static void validateImp(Imp imp) throws PreBidException {
+        final Banner banner = imp.getBanner();
+        final boolean isBannerSizesPresent = banner != null
+                && (ObjectUtils.anyNotNull(banner.getW(), banner.getH())
+                || CollectionUtils.isNotEmpty(banner.getFormat()));
```

Comment:
For this let's create method `isBannerSizesPresent` and invoke it directly in if statement